pkg/resources/ClusterUpdater.go

```diff
@@ -10,13 +10,13 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func ClusterUpdater() *schema.Resource {
+func ClusterUpdater(providerVersion string) *schema.Resource {
 	return &schema.Resource{
 		CreateWithoutTimeout: ClusterUpdaterCreateOrUpdate,
 		ReadContext:          schema.NoopContext,
```

Comment:
Why not implement `Read`, set the version there, and let terraform do the rest ?

Reply:
I tried that once and it seemed to not update on changes then. I might have done something wrong though. 😄

Reply:
@argoyle https://github.com/eddycharly/terraform-provider-kops/pull/607 WDYT ?

Reply:
This matches the outcome when I tried the Read-implementation myself as well. 😊

Reply:
Right, wrong logic, we can't refresh the state on read...
Pushed another commit, hope this is better :crossed_fingers:

Reply:
Yeay! When I finally managed to understand how the ldflags should be set when the variable is not in main it seems to work correctly.

You need to update the ldflags in goreleaser-config again for it to work though:
```
    ldflags:
      - -s -w
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildVersion={{.Version}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildCommit={{.Commit}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildDate={{.Date}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuiltBy=goreleaser'
```

I also change the build make-target to be able to do `make PROVIDER_VERSION=1.23.5`
```
.PHONY: build
build: gen
	@CGO_ENABLED=0 go build -ldflags="-s -w -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildVersion=v${PROVIDER_VERSION}'" ./cmd/terraform-provider-kops
```

Reply:
Thanks for cracking it down ! :+1: 
I updated the PR and will release one more alpha.